benchmarks/Benchmarks/BenchmarksRunner/Benchmarks.swift

```diff
@@ -0,0 +1,49 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftCertificates open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftCertificates project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftCertificates project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Benchmark
+import Benchmarks
+import Foundation
+
+let benchmarks = {
+    Benchmark.defaultConfiguration = .init(
+        metrics: [.mallocCountTotal, .syscalls] + .arc,
+        warmupIterations: 1
+    )
+
+    Benchmark("Verifier", configuration: .init(metrics: [.mallocCountTotal, .syscalls])) { benchmark in
+        for _ in benchmark.scaledIterations {
+            await verifier()
+        }
+    }
+
+    let runParseWebPKIRoots = parseWebPKIRoots()
+    Benchmark("Parse WebPKI Roots") { benchmark in
+        for _ in benchmark.scaledIterations {
+            runParseWebPKIRoots()
+        }
+    }
```

Comment:
I think the pattern for doing setup _inside_ the closure might be cleaner because you don't need to worry about conflicts in the outer scope. You do this by deferring the start of the measurement like this, IIUC:

```suggestion
    Benchmark("Parse WebPKI Roots") { benchmark in
        let runParseWebPKIRoots = parseWebPKIRoots()
        benchmark.startMeasurement()
        for _ in benchmark.scaledIterations {
            runParseWebPKIRoots()
        }
    }
```

Reply:
I'd suggest instead that you consider:

```suggestion
    Benchmark("Parse WebPKI Roots") { benchmark in
        for _ in benchmark.scaledIterations {
            runParseWebPKIRoots()
        }
    } setup: {
       parseWebPKIRoots()
    } 
```

Otherwise it'd be run for each iteration of the benchmark.

`scaledIterations` is the inner loop of the benchmark, but each benchmark function can also be called multiple times depending on run time and number of desired iterations, so using custom setup/teardown is intended to support such use cases. More details in [the documentation](https://swiftpackageindex.com/ordo-one/package-benchmark/1.9.1/documentation/benchmark/writingbenchmarks)

Reply:
The `setup` closure would need to allow me to return a value, in this case the `runParseWebPKIRoots` closure,  that is then passed into the benchmark function so I can actually run the benchmark. Otherwise I need to use some global storage.

Reply:
Ok, maybe it would make sense to have some state holder for the benchmark that can be accessed from setup/teardown to support that use case to avoid having to rerun it in the preamble.

Reply:
The setup closure can now return its state: https://github.com/ordo-one/package-benchmark/releases/tag/1.10.0